src/bin/ion/commands/generate/utils.rs

```diff
@@ -62,12 +23,24 @@ pub trait Language {
 
     /// Provides given target type as sequence
     /// e.g.
-    ///     target_type = "Foo" returns "ArrayList<Foo>"
+    ///     target_type = "Foo" returns "java.util.ArrayList<Foo>"
     ///     target_type = "Foo" returns "Vec<Foo>"
-    fn target_type_as_sequence(target_type: &Option<String>) -> Option<String>;
-
-    /// Returns true if the type name specified is provided by the target language implementation
-    fn is_built_in_type(name: &str) -> bool;
+    fn target_type_as_sequence(
+        target_type: FullyQualifiedTypeReference,
+    ) -> FullyQualifiedTypeReference;
+
+    /// Returns true if the type `String` specified is provided by the target language implementation
+    fn is_built_in_type(name: String) -> bool;
```

Comment:
This is a utility method used for a [tera filter](https://docs.rs/tera/1.19.0/tera/struct.Tera.html#method.register_filter). Since tera converts Rust data types into `tera::Value` which then would require to convert to `FullyQualifiedTypeReference` and then check if the type is built-in or not. Instead I have used the `string` representation itself here. But either way we do need to perform conversion from `tera::Value` to `String` or `FullyQualifiedTyepReference`.